okapi/services/attrs/attribute-definitions.xml

```diff
@@ -206,10 +225,11 @@ for the original implementations.
     </attr>
 
     <attr acode="A2" categories="de-cache-types">
-        <ocgs id="117" inc="true" />
-        <opencaching schema="OCPL" id="54" />
-        <opencaching schema="OCNL" id="54" />
-        <opencaching schema="OCRO" id="54" />
+        <ocgs id="208" inc="true" />
```

Comment:
OCGS IDs must not be changed. This would break backward compatibility with applications that already rely on the existing IDs.

Reply:
What would the actual real world impact (in the view of what is said in the comment just above)?

Reply:
Any application which processes OC specific attributes from GPX files will no longer recognize them. E.g. GASK will no longer show the OConly attribute, if 106 would be remapped.

Reply:
Also, OKAPI GPX would no longer be compatible with OCDE GPX.

Reply:
As I said before, I strongly recommend we urge third party app developers to adopt the new numbering scheme. 
AFAIK GSAK is a special case due to Clyde's illness.
Also AFAIK most popular geocaching apps that support OC have only partial support. 
Perhaps in part due to lack of documentation (until now?)

Reply:
I am willing to submit a DB and code update to OCDE along these lines. 
Would OCDE be willing to merge such a pull request? 
Then GPX compatibility won't become a problem.

As you know I am trying to bring some order to a lot of chaos and whatever scenario one considers ofcourse there will be pros and cons. IMO the best solution is to have a well defined set of data, given the circumstances with no overlaps and by design ensuring minimal chances of future overlaps and have it uniformly deployed and well documented. Sometimes it is best to move on rather then clinging to an old and chaotic scheme that is a nightmare to maintain. 
Just as an example: the Linux kernel developers dropped i386 and i486 CPU support at some point.

Reply:
Backward compatibility has always been a core principle of OKAPI. We would never break backward compatibility unless we are absolutely forced to do so.

OCDE has been exporting those IDs for almost 10 years now. There is no need to change them and break apps. They don't need any mainenance. They don't need to be "beautiful" in any way. They have been defined once, and noone ever will have to care again about them. 

New attributes are added very, very rarely. For the case this is done, I already documented which OCGS IDs are free and can be used for new attribs. This is all documented [here](https://github.com/opencaching/opencaching-pl/blob/4415114673e8dc713d844236e970cf3f134defba/lib/format.gpx.inc.php#L342). OCGS IDs for the special UK and US attributes are also defined there. Please use those, instead of trashing al the work the I already did.

Reply:
Understood. I will resequence the numbers with OCGS IDs as instructed.

However, with all due respect, as internal attrib IDs (independent from OCGS numbering) at each ocpl-based node will switch to a unified numbering and further attrib changes at UK, US, NL and RO (as requested by each), _some code will simply clean itself up_ because there won't be any further need to handle exceptions and exceptions to exceptions. It nothing about "trashing" anything and no ill intent.

Reply:
Will return early next week with modified proposal.

Reply:
As said, OCDE has been around for 10+ years, however the OCGS concept was introduced to OKAPI in 2017, so it is quite newer. 

I've checked the OKAPI code and, as far as I can tell so far, OCGS elements do not generate new elements in GPX export, instead they fill out <groundspeak:attribute elements and that's it. ( https://github.com/opencaching/okapi/issues/271 )
Same goes for OCDE GPX export and OCPL GPX export. 
I've also read several previous issues and discussions at OKAPI and gpx-extension-v1. 

AFAIK GSAK implements more extensively only OC-only (as attrib 106), including it in the filter dialog. Other OC specific attributes are displayed for the cache (taken from OCDE) but exported in GPX as ID only. 

My proposal does not make any changes to OC specific attributes that exist at OCDE (either both OCDE&OCPL or OCDE only). Thus any GPX exported by OKAPI or by OCDE code will not be affected. 

There seems to be very little support for OC attributes by third party apps and it seems to be more concentrated towards OCDE.

OCPL is in the long (and slow) process of implementing data definitions in the code to have a common base across installations (aiming to have in database only actual data, not data definitions; list of possible attributes counts as data definition). This cannot be done properly while each OCPL site has the same attribute with a different ID. 
Thus overhauling OCPL attributes is a critical step in that direction and a badly needed cleanup in a lot of chaotical data. It also simplifies the GPX export code.

My proposal changes only OCPL specific attributes with OCGS numbers in sync with OCPL internal ID numbers, assigned, I admit, to a different scheme then described here https://github.com/opencaching/opencaching-pl/blob/4415114673e8dc713d844236e970cf3f134defba/lib/format.gpx.inc.php#L342 ; 
Since even that numbering is non-continous, jumpy and with holes, in order to do the best to avoid further confusion, this proposal uses, for OCPL only, a distinct numbering set, compact, continuous, non-overlapping.

With OCGS IDs set as such and with OCPL code updates, with OCDE unaffected, GPX exports between OKAPI and OCPL code will also remain in sync, with same IDs.

I am also working with c:geo developers to extend their previously sketchy OC attribute support to improve it, greatly helped now by the documentation at https://wiki.opencaching.eu/index.php?title=Cache_attributes .

Reply:
OCDE was founded in 2005. The OCGS concept was introduced at OCDE around 2010. It was added to OCPL in 2017, and to OKAPI in 2018.

Reply:
Hence it is quite new from OCPL perspective. Also attributes for OCPL sites in OKAPI were incomplete and inconsistent. 
Taking into consideration that this is the first time since 2014 (when I first got involved in OCPL) that attributes for OCPL sites in attribute-definitions.xml are 100% consistent (with this PR) and documented and that OCDE data is untouched, I respectfully request to allow OCPL to move forward.

Reply:
If I shall do this, it needs more time. I try to 100% review and do some tests before merge. My OCPL devel VM is broken since a hard disk crash ~2 weeks ago, need to repair that first.

Reply:
Thank you. It is very wise to do so. 
Please bear in mind that this PR is designed to be deployed together with this [PR](https://github.com/opencaching/opencaching-pl/pull/2069) at opencaching-pl which includes per-site database content updates.